### ISSUE TITLE
[Feature Request] Self-Hosted Container

### ISSUE THREAD

**Issue by ghost-of-cerberus** [joelacus/Cobalt#3]
Greatly appreciate having Cobalt tab as a browser extension for Firefox, Microsoft Edge, and Chrome but would like to see a self-hosted container option. 

It appears that with the extension active, opening a new tab incurs an approximate 1.5 second loading delay as the extension renders the page. Most times I don't need the page to load as I just need an empty tab but the brief delay and loading of the page distracts me from what I was intending to do, incurring a time loss in my activities. With a container, the page would be rendered once by the webserver until/unless settings are changed and there would be no time to wait for the page to (re-)render.

Though, this may all be moot if render performance increases and the page load is < 1sec. Have to remember it's still beta. 😃 

For reference, my machine is an ASUS Zephyrus G14 (2020) 4900HS, 16GB RAM, nVIDIA RTX 2060 Max-Q, Windows 11 Pro (22H2), so wouldn't think it an issue with my hardware.

**Comment by joelacus** [comment]
This is a fair request, I have thought to myself the load times should be faster as well. I don't really know much about making containers, but I'll look into how I can do it. Would Docker containers be best?

In the mean time I will try to reduce the load times if I can.

Thank you for your feedback and support :)

**Comment by ghost-of-cerberus** [comment]
> Would Docker containers be best?

Yes. I would think that most container users are using Docker, though LXC/LXD and Podman are becoming more prevalent.


>  I don't really know much about making containers, but I'll look into how I can do it.

From what I can assume, Cobalt uses React and JS so a container with webserver "should" really be all you need. Would think an Nginx container to be the best option given speed and size.


> Thank you for your feedback and support :)

No worries. More than happy to help if and where needed. :)

**Comment by joelacus** [comment]
The extension is just pure javascript and jQuery. I am trying to learn React though and plan to rewrite Cobalt with it, but that will take a long time. Nevertheless, a container with a webserver sounds like the way to go. I do have an idea how to speed it up in general, but I'll look into the containers you mentioned.
